pagefaultgames/pokerogue
# [Bug Fix] Corrected PostSummonPhase Trigger Timing for Accurate Stat Changes

## What are the changes?
The opponent's PostSummonPhase was triggered before our Pokémon is on the field, preventing any stats change from our side (e.g., the ability Intimidate). The issue is that a wild Pokémon does not have a SummonPhase, so I moved it into the TurnInitPhase. For the player, it still triggers at the end of the SummonPhase. Additionally, I fixed the issue where the PostSummonPhase was trying to add the StatsChangePhase to the phase list at the end instead of just after the ShowAbility phase, causing it to occur way after the CommandPhase.

## Why am I doing these changes?
These changes were introduced to ensure that our Pokémon's abilities, like Intimidate, are properly applied when the Pokémon is summoned. This enhances the user experience by ensuring accurate stat changes. The issue was identified through gameplay and testing.

## What did change?
- **Before:** The opponent's PostSummonPhase was triggered before our Pokémon was on the field, so no stats change from our side could occur.
- **After:** The opponent's PostSummonPhase is moved to the TurnInitPhase for wild Pokémon and still triggers at the end of the SummonPhase for players.
- Fixed the issue where the PostSummonPhase was adding the StatsChangePhase at the end of the list instead of just after the ShowAbility phase.

## Screenshot
![image](https://github.com/pagefaultgames/pokerogue/assets/44787002/e3beb15b-14c0-4655-8caf-7d85e39db95a)

## How to test the changes?
- Check out the branch with these changes.
- Verify that the opponent's PostSummonPhase triggers correctly after our Pokémon is on the field.
- Confirm that the StatsChangePhase is added immediately after the ShowAbility phase.
- Run all unit tests to ensure they are passing (`npm run test`).

## Checklist
